### ISSUE TITLE
hugo command is working in local environement but shows errors with jenkins 

### ISSUE THREAD

**Issue by mahmoudozil** [thegeeklab/hugo-geekdoc#35]
![Capture](https://user-images.githubusercontent.com/26530104/101803109-772ebf80-3b10-11eb-802d-43831e6297ef.PNG)

Is there a difference between environments when webuild the hugo project?

**Comment by mahmoudozil** [comment]
executing "partials/head.html" at <index .Site.Data.assets "main.min.css">: error calling index: index of untyped nil

what's the meaning of this

**Comment by xoxys** [comment]
In general the environment doesn't matter. At least I'm not aware of any limitations. The error message indicates you are using a git clone instead of a release tarball download. It means that compiles assets are not found. That's mostly the case if you use a git clone as source, see https://geekdocs.de/usage/getting-started/#install-requirements

tl;dr 

> If you want to use the theme from a cloned branch instead of a release tarball you’ll need to install gulp locally and run the default pipeline once to create all required assets.

**Comment by mahmoudozil** [comment]
it works thank you. The problem like you said i used git clone for master instead of a release.

**Comment by alecthegeek** [comment]
I have this problem but I have installed from a release.  I checked the release into my project. Any suggestions? Should I install from GitHub?

**Comment by xoxys** [comment]
I've no idea so far. Do you use the same Hugo version locally and on Jenkins? Can you post the exact error output during the Jenkins run? Usually the environment does not really matter for Hugo builds.

**Comment by ebuildy** [comment]
@alecthegeek make sure in ``config.toml`` , theme is the right folder.

**Comment by osallou** [comment]
I have same issue with github actions. Works like a charm locally, with same hugo version

Build error link: https://github.com/bgo-bioimagerie/platformmanager/runs/3321956980?check_suite_focus=true

**Comment by xoxys** [comment]
@osallou Please dont hijack already closed issues in the future. Regarding your issue, it doesn't work locally as well:

```
git clone https://github.com/bgo-bioimagerie/platformmanager
Klone nach 'platformmanager' ...
remote: Enumerating objects: 12919, done.
remote: Counting objects: 100% (3051/3051), done.
remote: Compressing objects: 100% (1495/1495), done.
remote: Total 12919 (delta 2072), reused 2224 (delta 1493), pack-reused 9868
Empfange Objekte: 100% (12919/12919), 15.42 MiB | 9.51 MiB/s, fertig.
Löse Unterschiede auf: 100% (8679/8679), fertig.
[~/.tmp] $ cd platformmanager/doc
[~/.tmp/platformmanager/doc] $ hugo server
Start building sites … 
ERROR 2021/08/13 14:59:22 render of "page" failed: execute of template failed: template: _default/single.html:9:7: executing "_default/single.html" at <partial "head/others" .>: error calling partial: "/home/rknet/rkau2905/.tmp/platformmanager/doc/themes/hugo-geekdoc/layouts/partials/head/others.html:1:16": execute of template failed: template: partials/head/others.html:1:16: executing "partials/head/others.html" at <index .Site.Data.assets "js/darkmode.min.js">: error calling index: index of untyped nil
ERROR 2021/08/13 14:59:22 render of "page" failed: execute of template failed: template: _default/single.html:9:7: executing "_default/single.html" at <partial "head/others" .>: error calling partial: "/home/rknet/rkau2905/.tmp/platformmanager/doc/themes/hugo-geekdoc/layouts/partials/head/others.html:1:16": execute of template failed: template: partials/head/others.html:1:16: executing "partials/head/others.html" at <index .Site.Data.assets "js/darkmode.min.js">: error calling index: index of untyped nil
ERROR 2021/08/13 14:59:22 render of "page" failed: execute of template failed: template: _default/single.html:9:7: executing "_default/single.html" at <partial "head/others" .>: error calling partial: "/home/rknet/rkau2905/.tmp/platformmanager/doc/themes/hugo-geekdoc/layouts/partials/head/others.html:1:16": execute of template failed: template: partials/head/others.html:1:16: executing "partials/head/others.html" at <index .Site.Data.assets "js/darkmode.min.js">: error calling index: index of untyped nil
ERROR 2021/08/13 14:59:22 render of "page" failed: execute of template failed: template: _default/single.html:9:7: executing "_default/single.html" at <partial "head/others" .>: error calling partial: "/home/rknet/rkau2905/.tmp/platformmanager/doc/themes/hugo-geekdoc/layouts/partials/head/others.html:1:16": execute of template failed: template: partials/head/others.html:1:16: executing "partials/head/others.html" at <index .Site.Data.assets "js/darkmode.min.js">: error calling index: index of untyped nil
Built in 42 ms
```

You have not installed the theme properly, as e.g. the `data` directory is missing inside the `themes/hugo-geekdoc/` folder. 

```
[~/.tmp/platformmanager/doc] $ tree -L 1 themes/hugo-geekdoc/
themes/hugo-geekdoc/
├── archetypes
├── assets
├── images
├── layouts
├── LICENSE
├── README.md
├── static
├── theme.toml
└── VERSION

5 directories, 4 files

[~/.tmp/platformmanager/doc] $ tree -L 1 ~/Downloads/hugo-geekdoc
/home/rknet/rkau2905/Downloads/hugo-geekdoc
├── archetypes
├── assets
├── data
├── images
├── layouts
├── LICENSE
├── README.md
├── static
├── theme.toml
└── VERSION

6 directories, 4 files
```

**Comment by xoxys** [comment]
@osallou I've noticed that you use `--minify` in your GH Action, please read the docs https://geekdocs.de/usage/getting-started/#known-limitations

**Comment by osallou** [comment]
oops really sorry, did not see the *close*, other comments were added after the "close" event.
and thanks for info

**Comment by xoxys** [comment]
No worries 🙂
